tritium-registry/src/main/java/com/palantir/tritium/metrics/registry/TaggedMetricRegistry.java

```diff
@@ -70,6 +87,22 @@
     // This differs from MetricRegistry and takes the Gauge directly rather than a Supplier<Gauge>
     <T> Gauge<T> gauge(MetricName metricName, Gauge<T> gauge);
 
+    /**
+     * Registers and returns the specified gauge, replacing any existing gauge with the specified metric name.
+     *
+     * @param metricName metric name
+     * @param gauge gauge
+     */
+    // This differs from MetricRegistry and takes the Gauge directly rather than a Supplier<Gauge>
+    default <T> void registerOrReplaceGauge(MetricName metricName, Gauge<T> gauge) {
```

Comment:
Not sure if the type parameter buys us anything, but I don't have much preference either way:
```suggestion
    default void registerOrReplaceGauge(MetricName metricName, Gauge<?> gauge) {
```

Reply:
I think we should either log a warning when gauges are replaced, or return an ignorable boolean to allow callers to warn. Thoughts?